### ISSUE TITLE
`FutureOr`, `void`, and `dynamic` docs improvments

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/dartdoc#3678]
This may not be worth fixing, but [in the API docs](https://api.dart.dev/stable/3.3.0/dart-async/FutureOr-class.html) `FutureOr` is listed as being a "class", even though it isn't actually one. It is weird though because in the code, it is listed as an abstract class (with docs saying it really isn't a class). So presumably that is why it shows up as a class in dartdoc. This could be confusing, because there really are differences between just being a type and being a class.

In thinking about any other weird types like this, `void` and `dynamic` came to mind. Neither of these have any documentation at all that I can find, because we didn't do the `abstract class` workaround just to get docs.

It might be worth creating a new category called just "type", which we could use just to document these, with some special casing in dartdoc?

**Comment by srawlins** [comment]
Yeah, I'd love to get that all fixed; the main issue is https://github.com/dart-lang/dartdoc/issues/2391.

We could special case FutureOr though, to indicate it is not a class.

**Comment by jakemac53** [comment]
SGTM, feel free to close this as a dupe (originally I searched just for FutureOr issues).
